internal/tests_config.go

```diff
@@ -71,30 +40,9 @@ type TestsConfig struct {
 
 // Tests are the tests from yaml category pass / fail / skip.
 type Tests struct {
-	TestNames []string // tests names (i.e. "go.mongodb.org/mongo-driver/mongo/...")
-	OutRegex  []string // regexps that match the tests output (i.e. "^server version \"5.0.9\" is (lower|higher).*")
-}
-
-func (tc *TestsConfig) toMap() (map[string]status, error) {
-	res := make(map[string]status, len(tc.Pass.TestNames)+len(tc.Skip.TestNames)+len(tc.Fail.TestNames))
-
-	for _, tcat := range []struct {
-		testsStatus status
-		tests       Tests
-	}{
-		{Pass, tc.Pass},
-		{Skip, tc.Skip},
-		{Fail, tc.Fail},
-	} {
-		for _, t := range tcat.tests.TestNames {
-			if _, ok := res[t]; ok {
-				return nil, fmt.Errorf("duplicate test or prefix: %q", t)
-			}
-			res[t] = tcat.testsStatus
-		}
-	}
-
-	return res, nil
+	Names              []string // names (i.e. "go.mongodb.org/mongo-driver/mongo/...")
+	NameRegexPattern   []string // regex: "mongo.org/.*", the regex for the test name.
+	OutputRegexPattern []string // output_regex: "^server version \"5.0.9\" is (lower|higher).*".
```

Comment:
Is it a regex for test output? 
Then why we should fail if both `NameRegexPattern` and `OutputRegexPattern` match?

Reply:
https://github.com/FerretDB/dance/issues/169#issuecomment-1180519209